File: 김금환/완전탐색/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    int cnt1 = 0, cnt2 = 0, cnt3 = 0;
    int maxCnt = 0;
    
    for (int i = 0; i < answers.size(); ++i) {
        // 1번 수포자 확인
        if (answers[i] == ((i % 5) + 1)) {
            ++cnt1;
        }
        
        // 2번 수포자 확인
        if (answers[i] == 2 && (i % 2 == 0)) {
            ++cnt2;
        } else if (answers[i] == 1 && (i % 8) == 1) {
            ++cnt2;
        } else if (answers[i] == 3 && (i % 8) == 3) {
            ++cnt2;
        } else if (answers[i] == 4 && (i % 8) == 5) {
            ++cnt2;
        } else if (answers[i] == 5 && (i % 8) == 7) {
            ++cnt2;
        }
        
        // 3번 수포자 확인
        if (answers[i] == 3 && (i % 10 == 0 || i % 10 == 1)) {
            ++cnt3;
        } else if (answers[i] == 1 && (i % 10 == 2 || i % 10 == 3)) {
            ++cnt3;
        } else if (answers[i] == 2 && (i % 10 == 4 || i % 10 == 5)) {
            ++cnt3;
        } else if (answers[i] == 4 && (i % 10 == 6 || i % 10 == 7)) {
            ++cnt3;
        } else if (answers[i] == 5 && (i % 10 == 8 || i % 10 == 9)) {
            ++cnt3;
        }
        
        maxCnt = max({cnt1, cnt2, cnt3});
    }
    
    vector<int> answer;
    if (cnt1 == maxCnt) answer.push_back(1);
    if (cnt2 == maxCnt) answer.push_back(2);
    if (cnt3 == maxCnt) answer.push_back(3);
    return answer;
}

File: 이예진/완전탐색/모의고사.py
def solution(answers):
    answer = []
    l1 = [1,2,3,4,5]
    l2 = [2,1,2,3,2,4,2,5]
    l3 = [3,3,1,1,2,2,4,4,5,5]
    
    cnt1 = 0
    cnt2 = 0
    cnt3 = 0
    # l1,l2,l3을 무한반복하는 거보단 나머지로 자름
    for i in range(len(answers)):
        if l1[i%len(l1)] == answers[i]:
            cnt1 += 1
        if l2[i%len(l2)] == answers[i]:
            cnt2 += 1
        if l3[i%len(l3)] == answers[i]:
            cnt3 += 1
    
    l = [cnt1,cnt2,cnt3]
    for i in range(3):
        if l[i] == max(l):
            answer.append(i+1)

    
    return answer